cni/network/invoker_cns_test.go

```diff
@@ -310,13 +581,136 @@ func TestCNSIPAMInvoker_Delete(t *testing.T) {
 	}
 }
 
+func TestCNSIPAMInvoker_Delete_NotSupportedAPI(t *testing.T) {
+	require := require.New(t) //nolint further usage of require without passing t
+	// set new CNS API is not supported
+	unsupportedAPIs := make(map[cnsAPIName]struct{})
+	unsupportedAPIs["ReleaseIPs"] = struct{}{}
+
+	type fields struct {
+		podName      string
+		podNamespace string
+		cnsClient    cnsclient
+	}
+	type args struct {
+		address *net.IPNet
+		nwCfg   *cni.NetworkConfig
+		args    *cniSkel.CmdArgs
+		options map[string]interface{}
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "test delete happy path with unsupportedAPI",
+			fields: fields{
+				podName:      testPodInfo.PodName,
+				podNamespace: testPodInfo.PodNamespace,
+				cnsClient: &MockCNSClient{
+					unsupportedAPIs: unsupportedAPIs,
+					require:         require,
+					releaseIP: releaseIPHandler{
+						ipconfigArgument: getTestIPConfigRequest(),
+					},
+				},
+			},
+			args: args{
+				nwCfg: nil,
+				args: &cniSkel.CmdArgs{
+					ContainerID: "testcontainerid",
+					Netns:       "testnetns",
+					IfName:      "testifname",
+				},
+				options: map[string]interface{}{},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			invoker := &CNSIPAMInvoker{
+				podName:      tt.fields.podName,
+				podNamespace: tt.fields.podNamespace,
+				cnsClient:    tt.fields.cnsClient,
+			}
+			err := invoker.Delete(tt.args.address, tt.args.nwCfg, tt.args.args, tt.args.options)
+			if tt.wantErr {
+				require.Error(err)
+			} else {
+				require.NoError(err)
+			}
+		})
+	}
+}
+
+func TestCNSIPAMInvoker_Delete_NotFound(t *testing.T) {
```

Comment:
same comment as above. Not sure what this test is testing.

Reply:
i guess its testing failure case when both new and old api fails. @paulyufan you should name it as "Test Delete with both new and old api to release ip fails"